dtfs-central-api/src/services/utilisation-report-data-validator/index.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  UTILISATION_REPORT_HEADERS,
+  UtilisationReportDataValidationError,
+  UtilisationReportCsvCellData,
+  UtilisationReportCsvRowData,
+} from '@ukef/dtfs2-common';
+import {
+  generateUkefFacilityIdError,
+  generateBaseCurrencyError,
+  generateFacilityUtilisationError,
+  generateTotalFeesAccruedError,
+  generateTotalFeesAccruedCurrencyError,
+  generateTotalFeesAccruedExchangeRateError,
+  generatePaymentCurrencyError,
+  generatePaymentExchangeRateError,
+  generateFeesPaidForThePeriodError,
+  generateFeesPaidForThePeriodCurrencyError,
+} from './utilisation-report-cell-validators';
+
+const HEADER_IS_MISSING_BASE_ERROR_MESSAGE = 'header is missing or spelt incorrectly';
+
+/**
+ * Constructs header is missing error message
+ * @param header - The header text
+ * @returns The header is missing error message for the provided header
+ */
+export const getHeaderIsMissingErrorMessage = (header: string) => `${header} ${HEADER_IS_MISSING_BASE_ERROR_MESSAGE}`;
+
+/**
+ * Validate utilisation report csv headers
+ * @param csvDataRow - A row of data
+ * @returns Errors for any missing required headers and a list of the present headers
+ */
+export const validateUtilisationReportCsvHeaders = (
```

Comment:
should present headers also be returned in this function? as the function is called validate

Reply:
I am worried the name would be too long if we make it say `validateUtilisationReportCsvHeadersAndReturnAvailableHeaders` but if we move that logic out then we would be duplicating some of the logic and it could fall out of sync so here I'd been keen to keep it as it is. I have updated the docstring to add more detail.